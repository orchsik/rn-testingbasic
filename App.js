import React from 'react';
import {StyleSheet, View} from 'react-native';

import Profile from './src/Profile';

const App = ({userName, name}) => {
  return (
    <View style={styles.container}>
      <Profile userName={userName || '액션가면'} name={name || '짱구'} />
    </View>
  );
};

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
