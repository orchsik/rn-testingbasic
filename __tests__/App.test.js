import React from 'react';
import {render} from '@testing-library/react-native';

import App from '../App';

describe('[App] render', () => {
  const props = {userName: '액션가면', name: '짱구'};
  const rendered = render(<App {...props} />);

  test('renders correctly', () => {
    expect(rendered).toMatchSnapshot(); // 기존스냅샷과 일치하는지.
    expect(rendered).toBeTruthy(); // 컴포넌트가 null, undefined와 같은 falsy한 값을 가지는지
  });

  // test('should have default userName', () => {
  //   const rendered = render(<App />);
  // });
});

/**
 * App.js 컴포넌트를 불러와서 우리가 의도한 props를 주입하여 스냅샷을 비교하는 테스트.
 * 스냅샷 테스팅은 UI가 예측 못 하게 바뀌는걸 방지하도록할 때 유용한 툴.
 */
