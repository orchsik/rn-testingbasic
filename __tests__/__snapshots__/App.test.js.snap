// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[App] render renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      액션가면 X 짱구
    </Text>
    <Text
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      Select your status
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Bye!
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Hello!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
